lib/src/objdetect/objdetect_async.dart

```diff
@@ -0,0 +1,806 @@
+import 'dart:typed_data';
+import 'package:ffi/ffi.dart';
+import '../core/base.dart';
+import '../core/mat.dart';
+import '../core/point.dart';
+import '../core/rect.dart';
+import '../core/size.dart';
+import '../core/vec.dart';
+import '../opencv.g.dart' as cvg;
+
+class CascadeClassifier extends CvStruct<cvg.CascadeClassifier> {
+  CascadeClassifier._(cvg.CascadeClassifierPtr ptr, [bool attach = true])
+      : super.fromPointer(ptr) {
+    if (attach) {
+      finalizer.attach(this, ptr.cast(), detach: this);
+    }
+  }
+
+  factory CascadeClassifier.empty() {
+    final p = calloc<cvg.CascadeClassifier>();
+    cvRun(() => CFFI.CascadeClassifier_New(p));
+    return CascadeClassifier._(p);
+  }
+
+  factory CascadeClassifier.fromFile(String filename) {
+    final p = calloc<cvg.CascadeClassifier>();
+    final cp = filename.toNativeUtf8().cast<ffi.Char>();
+    cvRun(() => CFFI.CascadeClassifier_NewFromFile(cp, p));
+    calloc.free(cp);
+    return CascadeClassifier._(p);
+  }
+
+  Future<bool> loadAsync(String name) async {
+    final cname = name.toNativeUtf8().cast<ffi.Char>();
+    final rval = cvRunAsync<bool>(
+        (callback) => CFFI.CascadeClassifier_Load_Async(ref, cname, callback),
+        (c, p) {
+      final rval = p.cast<ffi.Int>().value != 0;
+      calloc.free(cname);
+      return c.complete(rval);
+    });
+    return rval;
+  }
+
+  Future<VecRect> detectMultiScaleAsync(InputArray image,
+      {double scaleFactor = 1.1,
+      int minNeighbors = 3,
+      int flags = 0,
+      (int, int) minSize = (0, 0),
+      (int, int) maxSize = (0, 0)}) async {
+    final ret = calloc<cvg.VecRect>();
+    final rval = cvRunAsync<VecRect>(
+        (callback) => CFFI.CascadeClassifier_DetectMultiScaleWithParams_Async(
+            ref,
+            image.ref,
+            scaleFactor,
+            minNeighbors,
+            flags,
+            minSize.cvd.ref,
+            maxSize.cvd.ref,
+            callback), (c, _) {
+      return c.complete(VecRect.fromPointer(ret));
+    });
+    return rval;
+  }
+
+  Future<(VecRect objects, VecInt numDetections)> detectMultiScale2Async(
+      InputArray image,
+      {double scaleFactor = 1.1,
+      int minNeighbors = 3,
+      int flags = 0,
+      (int, int) minSize = (0, 0),
+      (int, int) maxSize = (0, 0)}) async {
+    final ret = calloc<cvg.VecRect>();
+    final pnums = calloc<cvg.VecInt>();
+    final rval = cvRunAsync<(VecRect, VecInt)>(
+        (callback) => CFFI.CascadeClassifier_DetectMultiScale2_Async(
+            ref,
+            image.ref,
+            scaleFactor,
+            minNeighbors,
+            flags,
+            minSize.cvd.ref,
+            maxSize.cvd.ref,
+            ret,
+            pnums,
+            callback), (c, _) {
+      return c.complete((VecRect.fromPointer(ret), VecInt.fromPointer(pnums)));
+    });
+    return rval;
+  }
+
+    Future<(VecRect objects, VecInt numDetections, VecDouble levelWeights)> detectMultiScale3Async(
```

Comment:
here numDetections should be rejectLevels, if the line is too long, just omit names, -> `Future<(VecRect, VecInt, VecDouble)>`

Reply:
I will search how to update the project formatter configuration, since removing names will make code harder to debug I think